R/utils-co2_upper_lower.R

```diff
@@ -2,11 +2,33 @@ create_co2_range <- function(data) {
   data |>
     summarize_range(.data[[grep("co2_footprint", names(data), value = TRUE)]],
                     .by = c("grouped_by", "risk_category")) |>
-    jitter_range() |>
+    jitter_range(amount = 0.9) |>
+    stop_if_percent_noise_more_than_100() |>
     select(-c("min", "max")) |>
     rename(co2e_lower = "min_jitter", co2e_upper = "max_jitter")
 }
 
 add_co2_upper_lower <- function(data, co2_range) {
   left_join(data, co2_range, by = join_by("grouped_by", "risk_category"))
 }
+
+stop_if_percent_noise_more_than_100 <- function(data) {
+  check_noise <- data |>
+    mutate(max_percent_noise = percent_noise(.data$max, .data$max_jitter),
+           min_percent_noise = percent_noise(.data$min, .data$min_jitter))
+
+  bad_noise <- any(check_noise$max_percent_noise > 100) | any(check_noise$min_percent_noise > 100)
+  if (is.na(bad_noise)) {
+    rlang::warn(c(
+      "The noise data is `NA`",
+      i = "Are you using the correct data?"
+    ))
+    return(invisible(data))
+  }
```

Comment:
@kalashsinghal this fixes the mysterious failing check.

Reply:
Interesting case! Thanks a lot for resolving this issue! :)